src/main/java/com/jakub/todoSandbox/repository/InMemoryRepository.java

```diff
@@ -0,0 +1,165 @@
+                            + "\n priority: " + toBeSavedTodo.priority()
+                            + "\n steps: \n" + toBeSavedTodo.steps().get(0).name());
+            return toBeSavedTodo;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public void updateTodo(Long todoId, Todo todo) {
+        var exists = todoList.get(todoId);
+        if (exists != null) {
+            todoList.put(todoId, createNewTodoId(todoId, todo));
+            System.out.println("Todo using id: {" + todoId + "} has been updated.");
+        } else {
+            System.out.println("No todo exists under this id.");
+        }
+    }
+
+    @Override
+    public void deleteTodo(Long todoId) {
+        if (todoList.get(todoId) != null) {
+            todoList.remove(todoId);
+            System.out.println("Todo with the id: " + todoId + " was removed.");
+        } else {
+            System.out.println("There isn't any todo under id: " + todoId);
+        }
+    }
+
+    @Override
+    public void saveSteps(Long todoId, List<Step> createdSteps) {
+        var existingTodo = todoList.get(todoId);
+        if (existingTodo != null) {
+            System.out.println("Size of step array before adding new: " + existingTodo.steps().size());
+            for (Step step : createdSteps) {
+                if (canAddStepToTodo(existingTodo)) {
+                    Step createdStep = createNewStepForTodo(existingTodo, step);
+                    existingTodo.steps().add(createdStep);
+                } else {
+                    System.out.println("You have reached the maximum number of steps of 10 for todo with id: " + existingTodo.id());
+                    break;
+                }
+            }
+            System.out.println("Size of step array after adding new: " + existingTodo.steps().size());
+        } else {
+            System.out.println("Such a Todo does not exist.");
+        }
+    }
+
+    @Override
+    public void updateStep(Long todoId, int oldStepId, Step updatedStep) {
+        var existingTodo = todoList.get(todoId);
+        if (existingTodo != null) {
+            for (int i = 0; i < existingTodo.steps().size(); i++) {
+                if (existingTodo.steps().get(i).id() == oldStepId) {
+                    existingTodo.steps().set(i, createNewStepId(((long) oldStepId), updatedStep));
+                    System.out.println("Step with id:" + oldStepId + " has been updated");
+                    return;
+                }
+            }
+            System.out.println("Step with id:" + oldStepId + " does not exist");
+        } else {
+            System.out.println("Todo with id:" + todoId + " does not exist");
+        }
+    }
+
+    @Override
+    public void deleteSteps(Long todoId, List<Long> stepIds) {
+        var existingTodo = todoList.get(todoId);
+        if (existingTodo != null && stepIds.size() > 0 && !stepIds.contains(0L)) {
```

Comment:
Why do we need this `stepIds.size() > 0 && !stepIds.contains(0L)`? I don't think it is needed